### DISCUSSION TITLE
stdio重定向问题

### DISCUSSION THREAD

**Discussion by rocascent** [xmake-io/xmake#4527] [Q&A] [answered]
```lua
add_rules("mode.debug", "mode.release")

target("main")
    set_kind("binary")
    add_files("src/*.cpp")
    set_toolchains("clang")
    set_languages("cxx20")
    on_run(function(target)
        os.execv(target:targetfile(), {}, {stdin = "input.txt"})
    end)
```

我执行execv时重定向了输入，但是运行程序时不会输出到终端？

**Comment by waruqi** [comment]
会输出

```bash
xmake> os.execv("echo", {"hello"}, {stdin = "/tmp/a"})
hello
```

**Reply by rocascent** [reply]
等一下，我连echo都执行不了？
`error: cannot execv(echo ), No such file or directory`

**Reply by waruqi** [reply]
win下没有 echo 独立进程

**Reply by rocascent** [reply]
先不说这个。
`os.execv(target:targetfile(), {}, {stdin = "input.txt", stdout = "output.txt"})`
我设置了标准输出倒是能输出到文件，但是不加的话就不会输出到终端

**Reply by waruqi** [reply, marked as answer]
win 下才有的问题，这个地方导致 https://github.com/tboox/tbox/blob/059f19ed8f18cfcd59c1ca2f2aff3143157a0df4/src/tbox/platform/windows/process.c#L489

但是目前没有很好的办法解决，这块涉及很多细节东西 https://github.com/xmake-io/xmake/issues/3138#issuecomment-1338970250

**Comment by waruqi** [comment]
我开了个 issues 先备着，后面如果有时间可以再细看下。。https://github.com/xmake-io/xmake/issues/4546

**Comment by waruqi** [comment]
try this patch. please download installer here. https://github.com/xmake-io/xmake/actions/runs/19461944960

 https://github.com/xmake-io/xmake/pull/7035
